src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java

```diff
@@ -118,7 +122,7 @@ public boolean addMember(Member member) throws ParseException
         if (member.isAnnotationKey() && Kind.KIND_STRUCT != getKind())
         {
             throw new ParseException(null, "Error in member " + member.getName() +
-                    ": @key annotations only supported for structure's members.");
+                    ": @" + Annotation.key_str + " annotations only supported for structure's members.");
```

Comment:
I am not sure if something should be added to this message. The specification allows for discriminators to be declared as key too. Fast DDS-Gen does not support this yet, but the message could be interpreted as something that the specification does not allow or something that Fast DDS-Gen has not implemented yet.

```suggestion
                    ": @" + Annotation.key_str + " annotations only supported for structure's members (Union discriminator still pending implementation).");
```

Reply:
Done in 7a2f44f